src/types/profile-derived.js

```diff
@@ -101,8 +101,55 @@ export type Thread = {|
   // Strings for profiles are collected into a single table, and are referred to by
   // their index by other tables.
   stringTable: StringTable,
+  // The stack samples collected for this thread. This field is different from
+  // RawThread in that the `time` column is always present.
+  samples: SamplesTable,
 |};
 
+/**
+ * The derived samples table.
+ */
+export type SamplesTable = {|
```

Comment:
I think I would have used the same thing than before with `$Diff` to avoid some duplication, but I'm not sure it's better. `Omit` would be much easier to use but it's not in our version of Flow. So I'm fine with the duplication.

Reply:
I like the duplication in this case because it makes it clear that the types are completely independent and you don't need to worry about the other type when changing one of them.